README.md

```diff
@@ -1,13 +1,58 @@
-Python3 is required. Either use `python3` and `pip3` variants, or activate a Python virtual environment.
+# Installation
+Python3.5 is required. Either use `python3` and `pip3` variants, 
+or activate a Python virtual environment. If you are using the docker container `deepandreinforced/rl:latest`, you
+do not need to install this repo again. Otherwise:
+```bash
+python3 -m pip install -r requirements.txt
+python3 -m pip install -e .
+```
+# Goals
+This in an implementation of an OpenAI gym environment to train controllers 
+for Roboy. The goals are: 
+* To provide flexibility for you to train controllers for different 
+robots (e.g. MSJ Platfrom, Upper Body, etc.).
+* Second, it is straightforward to parallelize the environments s.t.
+you can scale the controller training with more compute resources.
+* Lastly, we kept it easy to implement communication to  your 
+simulation engine of choice (e.g. to MuJoCo, Gazebo, etc.).
+
+# Structure
+
+- The class `RoboyEnv` implements the `gym.GoalEnv` (OpenAI's interface definition).
+It is the central class in this repo.
+
+#### Directory `simulations/`
+- `SimulationClient` is a client interface to talk to our simulations. It provides
+the flexibility to change simulation engines by subclassing it.
+- `RosSimulationClient` is a subclass of `SimulationClient` that uses ROS services 
+to step and reset the simulation. We have used so far CARDSflow as our 
+simulation engine (not part of this repo).
+
+#### Directory `robots/`
+- `RoboyRobot` is the interface for any robot. You can add new robots
+by subclassing from it.
+- `MsjRobot` is a subclass of `RoboyRobot` with concrete dimensions and boundaries.
+
+#### Class Diagram
+The `RoboyEnv` uses a `SimulationClient` to read values from the simulation.
+The `SimulationClient` depends upon a `RoboyRobot` to parse the numbers it
+receives from the simulation into a meaningful `RobotState` 
+
+![Class Diagram](./docs/class-diagram.png)
 
-# gym-roboy
-pip install -e .
```

Comment:
The user do not to install gym-roboy anymore, because it is already installed in the docker?

Reply:
@ALPA-Industry-and-Technology Yes. We do not **need** to install it at all, because we are launching the training scripts directly from the `gym_roboy` package, so all the paths are resolved as relative paths.

Reply:
Yep, but we could mention pip package installation. The structure is suitable for the gym.make('msj-control-v1') which also mentioned in the documentation of gym environments.